EKF/terrain_estimator.cpp

```diff
@@ -99,7 +99,7 @@ void Ekf::fuseHagl()
 		_hagl_innov = pred_hagl - meas_hagl;
 
 		// calculate the observation variance adding the variance of the vehicles own height uncertainty and factoring in the effect of tilt on measurement error
-		float obs_variance = fmaxf(P[9][9], 0.0f) + sq(_params.range_noise / _R_rng_to_earth_2_2);
+		float obs_variance = fmaxf(P[9][9], 0.0f) + (sq(_params.range_noise) + sq(_params.range_noise_scaler * _range_sample_delayed.rng)) * _R_rng_to_earth_2_2;
```

Comment:
Actually, there is an error, but thats not it. If the vehicle is banked at 45 degrees, each metre of range error will only produce 0.71m of height error, so it should be *. However the _R_rng_to_earth_2_2_ should be squared, because it scales from range to height and we are calculating a variance that has units of m^2.